openvpn: add common client/server config section

dmonakhov opened the pull request:
Russian censorship agency (RKN) going crazy this week and blocked 16M of ip addresses. 
But anyone can setup openvpn server in kubernetes cluster. 

**LOG
By default server configured with obsoleted cryptography algo so
client complains about weak security:
 "WARNING: INSECURE cipher with block size less than 128 bit (64 bit). 
 This allows attacks like SWEET32.  Mitigate by using a --cipher with
 a larger block size (e.g. AES-256-CBC)."

Let's use AES-256-CBC by default so deployment becomes more secure out of box.

dmonakhov commented:
/assign @viglesiasce

viglesiasce commented:
/ok-to-test

viglesiasce commented:
cc @jfelten

jfelten commented:
I think this request is valid. Questions:
 
Why add a new genconf section instead of using the existing conf section?  Having 2 configurations that do the same thing doesn't make sense to me.

dmonakhov commented:
@jfelten I've added second option because originally 'config' represents server-speciffig config options (logging, max clients, etc). But it is also nice to have generic-config which will be appended to both  server and client, this part contains negotiation parameters (chiper, compression, etc) . That is why we need two configs. BTW I've added corresponding comment to values.yaml file

May be it is reasonable to rename 'config' to 'srvconfig' for convinience, but it try to preserve old names.

jfelten-isp commented:
@dmonakhov I am not sure I understand.  Both conf blocks are getting added to the openvpn.conf, which is used by the openvpn server.  Different server and client configs make sense, but as written both conf sets get added to the same place.  Did you mean to classify different settings  as eitehr server or client?  Did you want to add options to the generated client ovpn file?

dmonakhov commented:
@jfelten-isp My idea is follows:
'Values.openvpn.genconf' stands for GenericConfig so it is common part for client and server, it contains negotiation part ( chiper proto, compression proto, auth proto, special routing, etc) that is why I appended it to both configs:
- client's config via newClientCert.sh
- server config via server's openvpn.conf

'Values.openvpn.genconf' stands for server only options (log rotation, max client, etc)

jfelten-isp commented:
@dmonakhov - got it and good idea.  My only concern is how to explain that in documentation.  Can we either change the name genconf to something more intuitive like clientserverconf or add a note in the README?

Commit:
openvpn: add common client/server config section

By default server configured with obsoleted cryptography algo so
client complains about weak security:
 "WARNING: INSECURE cipher with block size less than 128 bit (64 bit). 
 This allows attacks like SWEET32.  Mitigate by using a --cipher with
 a larger block size (e.g. AES-256-CBC)."

Let's use AES-256-CBC by default so deployment becomes more secure out of box.

dmonakhov commented:
@jfelten-isp I've updated PR, options now called commonConfig and serverConfig. Readme also updated

viglesiasce commented:
@dmonakhov is it worth adding clientConfig now too? 

@jfelten @jfelten-isp WDYT?

jmymy commented:
any update on this?
